pkg/wait/wait_for_ready.go

```diff
@@ -178,6 +196,30 @@ func (w *waitForReadyConfig) waitForReadyCondition(start time.Time, name string,
 	}
 }
 
+// Wait until the expected EventDone is satisfied
+func (w *waitForEvent) Wait(name string, timeout time.Duration, msgCallback MessageCallback) (error, time.Duration) {
+	watcher, err := w.watchMaker(name, timeout)
+	if err != nil {
+		return err, 0
+	}
+	defer watcher.Stop()
+	start := time.Now()
+	// channel used to transport the error
+	errChan := make(chan error)
+	for {
+		select {
+		case <-time.After(timeout):
```

Comment:
Not sure if this is correct (i know, it the same as before, but still ;-), as it looks like that the timer starts afresh for every event that comes in. Shouldn't it be the timeout for the overall operation ? (so we would need to run the timer in a separate gofunc and communicate via channel, a bit like in the delaye case we have in waitForReady)

Reply:
I think, you're right. I had a feeling the timeout isn't working actually. Anyway I should add more test cases to unit tests.